src/opening-period/page/EditOpeningPeriod.tsx

```diff
@@ -0,0 +1,301 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { Control, Controller, FieldValues, useForm } from 'react-hook-form';
+import { Button, Notification, LoadingSpinner } from 'hds-react';
+import { useHistory } from 'react-router-dom';
+import api from '../../common/utils/api/api';
+import {
+  DatePeriod,
+  Language,
+  Resource,
+  ResourceState,
+} from '../../common/lib/types';
+import { transformToApiFormat } from '../../common/utils/date-time/format';
+import Datepicker from '../../components/datepicker/Datepicker';
+import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
+import { ResourceInfo } from '../../resource/page/ResourcePage';
+import './EditOpeningPeriod.scss';
+
+type FormData = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string;
+  openingPeriodEndDate: string;
+};
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+const FormDatePicker = ({
+  name,
+  label,
+  control,
+  register,
+}: {
+  name: string;
+  label: string;
+  control: Control<FormData & FieldValues>;
+  register: () => void;
+}): JSX.Element => (
+  <Controller
+    control={control}
+    name={name}
+    render={({ value, onChange }): JSX.Element => (
+      <Datepicker
+        id={name}
+        dataTest={name}
+        labelText={label}
+        onChange={onChange}
+        value={new Date(value)}
```

Comment:
Where does this value come from?

Reply:
it is the initial value of the react-hook-form which is undefined when defaultValues are not set